src/main/java/org/terasology/launcher/TerasologyLauncher.java

```diff
@@ -16,111 +16,86 @@
-import javafx.scene.layout.VBox;
-import javafx.stage.Screen;
 import javafx.stage.Stage;
-import javafx.stage.StageStyle;
-import javafx.util.Duration;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 import org.terasology.crashreporter.CrashReporter;
-import org.terasology.launcher.gui.javafx.ApplicationController;
+import org.terasology.launcher.gui.javafx.MainController;
 import org.terasology.launcher.log.TempLogFilePropertyDefiner;
 import org.terasology.launcher.util.BundleUtils;
 import org.terasology.launcher.util.Languages;
-import org.terasology.launcher.util.LauncherStartFailedException;
 import org.terasology.launcher.version.TerasologyLauncherVersionInfo;
 
 import java.io.IOException;
 import java.nio.file.Path;
 import java.util.Arrays;
 import java.util.List;
 import java.util.MissingResourceException;
-import java.util.ResourceBundle;
 
 public final class TerasologyLauncher extends Application {
 
     private static final Logger logger = LoggerFactory.getLogger(TerasologyLauncher.class);
 
-    private static final int SPLASH_WIDTH = 800;
-    private static final int SPLASH_HEIGHT = 223;
-
-    private Pane splashLayout;
-    private ProgressBar loadProgress;
-    private Label progressText;
-    private Stage mainStage;
-    private HostServices hostServices;
+    private final LauncherInitTask initTask = new LauncherInitTask();
+    private LauncherConfiguration initialConfig;
+    private MainController mainController;
 
     public static void main(String[] args) {
         launch(args);
     }
 
     @Override
-    public void init() throws Exception {
-        ImageView splash = new ImageView(BundleUtils.getFxImage("splash"));
-        loadProgress = new ProgressBar();
-        loadProgress.setPrefWidth(SPLASH_WIDTH);
-        progressText = new Label();
-        splashLayout = new VBox();
-        splashLayout.getChildren().addAll(splash, loadProgress, progressText);
-        progressText.setAlignment(Pos.CENTER);
-        splashLayout.getStylesheets().add(BundleUtils.getStylesheet("css_splash"));
-        splashLayout.setEffect(new DropShadow());
-        hostServices = getHostServices();
+    public void init() {
+        initTask.setOnSucceeded(event -> {
+            if (mainController != null) {
+                mainController.update(initTask.getValue());
+            } else {
+                initialConfig = initTask.getValue();
+            }
+        });
+        initTask.setOnFailed(event -> openCrashReporterAndExit((Exception) initTask.getException()));
+        new Thread(initTask).start();
     }
 
     @Override
-    public void start(final Stage initialStage) {
+    public void start(Stage mainStage) throws IOException {
         logger.info("TerasologyLauncher is starting");
-        logSystemInformation();
 
+        logSystemInformation();
         initProxy();
         initLanguage();
 
-        final Task<LauncherConfiguration> launcherInitTask = new LauncherInitTask(initialStage);
```

Comment:
By moving this to `init()`, the proxy and language init above will not be executed before the task is started. Thus, communication tasks might not have the correct proxy settings. Also, I'm not sure, how internationalization of error messages etc. works if the language settings are done afterwards.